lib/ipmi_hpmfwupg.c

```diff
@@ -1208,8 +1212,8 @@ HpmFwupgActionUploadFirmware(struct HpmfwupgComponentBitMask components,
 				if (rc == HPMFWUPG_UPLOAD_BLOCK_LENGTH && !bufLengthIsSet) {
 					rc = HPMFWUPG_SUCCESS;
 					/* Retry with a smaller buffer length */
-					if (strstr(intf->name,"lan") && bufLength > 8) {
-						bufLength-= 8;
+					if (strstr(intf->name,"lan") && bufLength > 2) {
+						bufLength >>= 1;
```

Comment:
I'd prefer `/= 2`. Let the compiler do optimization and make the code look closer to what it actually does.
A comment above the `if()` would be nice, explaining what is going on here.
Why do you think that reducing the size in half is a good approach (beside that it works) ?
If there is any justification for this algorithm with a reference to the HPM specification, it would be nice to have it in a comment. If not, then just add your own explanation of why you consider this a correct approach. Apparently, it doesn't find the maximum working block size. The old code attempted to to exactly that.

Reply:
Hi Alexander,
> I'd prefer `/= 2`. Let the compiler do optimization and make the code look closer to what it actually does.
Ocean: Sure!
> A comment above the `if()` would be nice, explaining what is going on here.
Ocean: Sure!
> Why do you think that reducing the size in half is a good approach (beside that it works) ?
Ocean: I just want to quickly find the valid size and ensure the upload not blocked.
> If there is any justification for this algorithm with a reference to the HPM specification, it would be nice to have it in a comment. If not, then just add your own explanation of why you consider this a correct approach. Apparently, it doesn't find the maximum working block size. The old code attempted to to exactly that.
Ocean: I check the PICMG® HPM.1 R1.0 and the block size depends on upgrade agent(here means ipmitool) itself: 
REQ 3.75 An Upgrade Agent shall select the block size in such a way that the “Upload firmware
block” command does not violate any protocol requirements (such as the maximum
message size defined for IPMB packets).
I am thinking a new algorithm to balance the maximum working block size and efficiency. I would update later.

Thanks for your review!
Ocean.